internal/handlemap/lru_policy.go

```diff
@@ -0,0 +1,254 @@
+package handlemap
+
+import (
+	"container/list"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+)
+
+type BlockKey struct {
+	StartIndex int64
+	Handle     *Handle
+}
+
+type CacheBlock struct {
+	sync.RWMutex
+	StartIndex int64
+	EndIndex   int64
+	Data       []byte
+	Last       bool // last block in the file?
+}
+
+type DiskBlock struct {
+	StartIndex int64
+	EndIndex   int64
+	Path       string
+}
+
+//LRUCache definition for Least Recently Used Cache implementation.
+type LRUCache struct {
+	Capacity        int64                      //defines a cache object of the specified capacity.
+	List            *list.List                 //DoublyLinkedList for backing the cache value.
+	Elements        map[BlockKey]*list.Element //blockKey: KeyPair
+	Occupied        int64
+	DiskPersistence bool
```

Comment:
we don't need this flag on per handle basis, it's a global thing at policy level.

Reply:
yea I had issues with imports so I placed this file here for now, we can discuss in technical meeting